pages/DSNP/Identity.md

```diff
@@ -19,13 +19,34 @@ A user's ownership of their identity is expressed via ownership and control of t
 Control entails the power to announce content associated with the identifier and
 the ability to delegate permission to others to announce content on the user's behalf.
 
+## Retirement
+
+A user may choose to retire their identity at any time.
+Once an identifier is retired, an implementation MAY remove all state data associated with that identifier, provided that an indication that the identifier is retired remains, so it may not be reused in the future.
+This means that all data previously sent from the identifier, the control keys associated with the identifier, and the delegations (see next section) associated with the identifier may be removed.
+
 ### Delegation
 
 * An owner MUST be able to delegate permission to announce on their behalf to other parties.
 * A user MUST be able to revoke delegated permissions.
 * Announcements from a delegate MUST be able to be verify which delegate made the specific announcement.
 * Delegation revocation MUST NOT be retroactive.
 
+## Operations
+
+Each invocation of a DSNP Operation MUST have verifiable approval of the acting principal(s) via signature proof or other system in the form of a Control Key Ownership Proof.
+For example, an implementation might use the public key of an asymmetric key pair as a control key, and provide a proof for each operation by producing a cryptographic signature of the user's DSNP Identifier and some nonce value.
+
+| Operation | Optional? | Principal(s) | Inputs | State Change Record |
+|---------- |---------- |------------- |------- |-------------- |
+| Create Identifier | no | User | Control Key, Control Key Ownership Proof | User Id Creation Record { User's Identifier, Control Key } |
+| Retire Identifier | no | User | None | User Id Retirement Record { User's Identifier } |
+| Define Delegation | no | User AND Delegate | User's Identifier, Delegate's Identifier, Set of Allowed  [Announcement Types](Announcements.md#announcement-types) | Delegation Definition Record { User's Identifier, Delegate's Identifier, Set of Allowed Announcement Types } |
+| Revoke Delegation | no | User OR Delegate | User's Identifier, Delegate's Identifier | Delegation Revocation Record { User's Identifier, Delegate's Identifier } |
+| Add Control Key | YES | User | Control Key, Control Key Ownership Proof | Control Key Addition Record { User's Identifier, Control Key } |
+| Remove Control Key | YES | User | Control Key | Control Key Removal Record { User's Identifier, Control Key } |
```

Comment:
Suggestion Only (Matter of visual syntax)

I don't quite get the `{}` syntax at the end of the State Change Record column. While it is trying to say these are the data items for the record, it is hard to parse in the format. I wonder if we want a different table of records or place to define each record to more clearly?

Perhaps another option is a subtable? (a bit gross for sure).

Another option might be separating out all the operations into a separate page and making each row a section? That would give a greater space to Inputs and State Change Record.

Reply:
Yes, I agree the format is a bit lacking.

How about this:
* A page listing all Operations
* A page listing all Record types
* A simple list of Operations (simple names with links) on Identity.md and Announcements.md

This provides an easier format (a checklist of sorts) for implementations to refer to and use as a template.

Reply:
OK I've updated this, let me know what you think.